src/fugw/scripts/coarse_to_fine.py

```diff
@@ -373,19 +373,17 @@ def fit(
     source_geometry_embeddings = _make_tensor(source_geometry_embeddings)
     target_geometry_embeddings = _make_tensor(target_geometry_embeddings)
 
-    # Compute anatomical kernels
-    source_geometry_kernel = torch.cdist(
-        source_geometry_embeddings[source_sample],
-        source_geometry_embeddings[source_sample],
-        p=2,
+    # Compute coarse geometry embeddings
+    source_coarse_embedding = source_geometry_embeddings[source_sample]
+    target_coarse_embedding = target_geometry_embeddings[target_sample]
+
+    # Normalize embeddings
+    source_coarse_embedding = (
+        source_coarse_embedding / source_coarse_embedding.max()
```

Comment:
This is a normalization  I guess ? You don't take the absolute value ?

Reply:
Yes my mistake the normalization should have been done with respect to the max norm of the geometry, not the embedding directly. This has been fixed.